[SIP-5] Refactor and repair partition

kristw opened the pull request:
- Extract `slice` and `formData` according to [SIP-5]
- Fix small issues and syntax inside heatmap.
- Fix broken tooltip: Add background, reorder columns into readable format. add padding between columns

@williaster @conglei 

Before
![icicle_test-2](https://user-images.githubusercontent.com/1659771/44601089-41733a80-a790-11e8-80ca-8ca7d0c3de39.png)

After
![icicle_test](https://user-images.githubusercontent.com/1659771/44601094-4637ee80-a790-11e8-99b4-2034f6f17a32.png)

kristw committed:
Extract slice and formData

kristw committed:
reorder functions

kristw committed:
fix tooltip

kristw committed:
remove commented code

williaster commented on superset/assets/src/visualizations/partition.js:
```diff
@@ -221,26 +250,52 @@ function partitionVis(slice, payload) {
+        );
+        t += (
+          '<tr class="emph">' +
+            '<td class="legend-color-guide" style="opacity: 0.75">' +
```
do styles for `.legend-color-guide` and `.x-value` come from another style sheet? or do they do anything?

williaster reviewed (approved):
couple comments, looks solid overall! 🎈

kristw commented on superset/assets/src/visualizations/partition.js:
```diff
@@ -324,10 +345,78 @@ function partitionVis(slice, payload) {
+    logScale,
+    metrics,
+    numberFormat,
+    partitionLimit: partitionLimit ? parseInt(partitionLimit, 10) : partitionLimit,
```
👍

kristw committed:
remove commented code and rename variables

kristw committed:
remove verboseMap

kristw committed:
rename kx, ky to zoomX, zoomY

kristw commented:
I addressed the comments above.
- Remove commented code
- Rename `kx` `ky`
- Remove legacy className from nvd3 tooltip
- Remove `verboseMap` from prop.
